### PR TITLE
Make all tests parallel where possible, added linter to look for parallel tests

### PR DESCRIPTION
**Pull request by James-Pickett**

### CODE REVIEW

File: pkg/log/checkpoint/files_test.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ func TestFilesFound(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run("testFilesFound", func(t *testing.T) {
+			t.Parallel()
```
Comment (directionless) [inline pkg/log/checkpoint/files_test.go:34]: Do we need to re-init tt here? We probably should, so we're consistent

File: pkg/make/builder_test.go [hunk 2]
```diff
@@ -53,6 +53,8 @@ func TestNamingHelpers(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run("platform="+tt.platform, func(t *testing.T) {
+			t.Parallel()
```
Comment (directionless) [inline pkg/make/builder_test.go:56]: re-init tt?

File: pkg/osquery/tables/munki/munki_test.go [hunk 3]
```diff
@@ -46,6 +46,8 @@ func TestGenerateMunkiReport(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (directionless) [inline pkg/osquery/tables/munki/munki_test.go:49]: re-init?

File: pkg/simulator/hosts_test.go [hunk 4]
```diff
@@ -164,6 +173,8 @@ func TestRunQuery(t *testing.T) {
 
 	for _, tt := range testCases {
 		t.Run("", func(t *testing.T) {
+			t.Parallel()
```
Comment (directionless) [inline pkg/simulator/hosts_test.go:176]: re-init?

File: pkg/log/checkpoint/files_test.go (hunk 1 above)
Comment (James-Pickett) [reply pkg/log/checkpoint/files_test.go:34]: This is a good one, the lack of re-init made the test give a false pass.

The all above to `defer os.RemoveAll(tempDir)` is being executed before/during the parallel subtests and was deleting them temp directory ... which makes sense. 

Discovered a nifty little `t.Cleanup(func ())` built into the testing package. 

"Cleanup registers a function to be called when the test and all its subtests complete. Cleanup functions will be called in last added, first called order."

```
	t.Cleanup(func() {
		os.RemoveAll(tempDir)
	})
```

**Review by directionless — APPROVED** [review]
Let's try! More parallel is more better!

**Pull request merged**